Classes/Controller/MediaController.php

```diff
@@ -45,8 +49,26 @@ public function videoAction() {
         $langId = $context->getPropertyFromAspect('language', 'id');
         $site = $GLOBALS['TYPO3_REQUEST']->getAttribute('site');
         $language = $site->getLanguageById($langId);
-        $langCode = $language->getTwoLetterIsoCode();
-        $this->view->assign('langCode', $langCode);
+        $locale = $language->getTwoLetterIsoCode();
+
+        // Prepare passing translation strings to the JS frontend code. First,
+        // grab the available translation keys from the translation base file
+        // (without language prefix); we treat this as authoritative on which
+        // keys exist in case a translated file (such as de.locallang_video.xlf)
+        // is missing or adding extra keys.
+        $localizationFactory = GeneralUtility::makeInstance(LocalizationFactory::class);
+        $baseTranslationFile = "EXT:{$this->extKey}/Resources/Private/Language/locallang_video.xlf";
```

Comment:
I don't like "this->extKey" anymore. The core has switched to just write your extension key down. In this case, it's twice in the file. Not that frequently. ;-)

Reply:
Me neither ;-) I changed it.